crates/miden-tx/tests/integration/scripts/compatibility.rs

```diff
@@ -0,0 +1,490 @@
+    let recipient = NoteRecipient::new(serial_num, note_script, NoteInputs::default());
+    let incompatible_custom_note = Note::new(vault, metadata, recipient);
+    assert_eq!(CheckResult::No, target_account_interface.can_consume(&incompatible_custom_note));
+
+    // STATEFUL CHECK
+    // --------------------------------------------------------------------------------------------
+
+    // TODO: implement
+}
+
+#[test]
+fn test_custom_account_multiple_components_custom_notes() {
+    // STATELESS CHECK
+    // --------------------------------------------------------------------------------------------
+
+    let account_custom_code_source = "
+        export.procedure_1
+            push.1.2.3.4 dropw
+        end
+
+        export.procedure_2
+            push.5.6.7.8 dropw
+        end
+    ";
+
+    let custom_component = AccountComponent::compile_with_path(
+        account_custom_code_source,
+        TransactionKernel::testing_assembler(),
+        vec![],
+        LibraryPath::new("test::account::component_1").unwrap(),
+    )
+    .unwrap()
+    .with_supports_all_types();
+
+    let mut rng = ChaCha20Rng::from_seed(Default::default());
+    let sec_key = SecretKey::with_rng(&mut rng);
+    let pub_key = sec_key.public_key();
+
+    let rpo_component = RpoFalcon512::new(pub_key);
+
+    let target_account = AccountBuilder::new(ChaCha20Rng::from_entropy().gen())
+        .with_component(custom_component.clone())
+        .with_component(BasicWallet)
+        .with_component(rpo_component)
+        .build_existing()
+        .unwrap();
+    let target_account_interface = AccountInterface::from(&target_account);
+
+    let mut mock_chain = MockChain::with_accounts(&[target_account.clone()]);
+    let sender_account =
+        mock_chain.add_existing_wallet(Auth::BasicAuth, vec![FungibleAsset::mock(20)]);
+    let serial_num =
+        RpoRandomCoin::new([Felt::new(1), Felt::new(2), Felt::new(3), Felt::new(4)]).draw_word();
+    let tag = NoteTag::from_account_id(target_account.id(), NoteExecutionMode::Local).unwrap();
+    let metadata = NoteMetadata::new(
+        sender_account.id(),
+        NoteType::Public,
+        tag,
+        NoteExecutionHint::always(),
+        Default::default(),
+    )
+    .unwrap();
+    let vault = NoteAssets::new(vec![FungibleAsset::mock(100)]).unwrap();
+
+    let compatible_source_code = "
+        use.miden::contracts::wallets::basic->wallet
+        use.miden::contracts::auth::basic->basic_auth
+        use.test::account::component_1->test_account
+
+        begin
+            push.1
+            if.true 
+                call.wallet::receive_asset
+                call.wallet::create_note
+                call.wallet::move_asset_to_note
+                call.test_account::procedure_1
+                call.test_account::procedure_2
+                call.basic_auth::auth_tx_rpo_falcon512
+            else
+                call.test_account::procedure_1
+                call.test_account::procedure_2
```

Comment:
This test is supposed to check that when multiple components are present, the result is "maybe" if the branches contain procedures from the account's components, right? Shouldn't we then have procedures from multiple components in _all_ branches, to not make this test "too easy", so to say?
So basically, can we add more valid calls to procedures in the else branch to make the test more complex? Otherwise it seems similar to the previous test.